internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,357 @@
+	if testing.Short() {
+		t.Skip("skipping testing due to unavailability of testing environment")
+	}
+	//include a failure case
+	dbcfg := config.DbCfg{
+		Port:      "9085",
+		Host:      "localhost",
+		Driver:    "mysql",
+		User:      "root",
+		Pass:      "pass",
+		DbName:    "useracc",
+		TableName: "newTemp",
+	}
+	dataBase := config.Connect(dbcfg, dbcfg.TableName)
+	svcConfig := config.SvcConfig{
+		DbSvc: config.DbSvc{Db: dataBase},
+	}
+	dB := NewSql(config.DbSvc(svcConfig.DbSvc), "newTemp")
+
+	tests := []struct {
+		name        string
+		setupFunc   func()
+		cleanupFunc func()
+		filter      map[string]interface{}
+		validator   func(bool)
+		dbInterface DataSourceI
+	}{
+		{
+			name: "SUCCESS::Health check",
+			validator: func(res bool) {
+				if res != true {
+					t.Errorf("Want: %v, Got: %v", true, res)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			res := dB.HealthCheck()
+
+			if tt.validator != nil {
+				tt.validator(res)
+			}
+		})
+	}
+}
+func TestGet(t *testing.T) {
+	tests := []struct {
+		name        string
+		setupFunc   func() sqlDs
+		cleanupFunc func()
+		filter      map[string]interface{}
+		validator   func([]model.Transaction, int, error)
+		dbInterface DataSourceI
+	}{
+		{
+			name: "SUCCESS::Get",
+			filter: map[string]interface{}{
+				"user_id":        "1234",
+				"account_number": 1,
+			},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE user_id = '1234' AND account_number = 1")).WillReturnError(nil).WillReturnRows(sqlmock.NewRows([]string{"count(transaction_id)"}).AddRow("1"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE user_id = '1234' AND account_number = 1 ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnRows(sqlmock.NewRows([]string{"transaction_id", "account_number", "user_id", "amount", "transfer_to", "created_at", "updated_at", "status", "type", "comment"}).AddRow("0000-1111-2222-3333", 1, "4444-1111-2222-3333", 1000, 1234567890, time.Now(), time.Now(), "approved", "debit", "no comments"))
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				temp := model.Transaction{
+					TransactionId: "0000-1111-2222-3333",
+					AccountNumber: 1,
+					UserId:        "4444-1111-2222-3333",
+					Amount:        1000,
+					TransferTo:    1234567890,
+					Status:        "approved",
+					Type:          "debit",
+					Comment:       "no comments",
+				}
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err)
+					return
+				}
+				if count != 1 {
+					t.Errorf("Want: %v, Got: %v", 3, count)
+				}
+				if !reflect.DeepEqual(rows[0].TransactionId, temp.TransactionId) {
```

Comment:
Create a variable of type []model.Transaction and populate the expected value(s) in it. The compare the got vs want.

Reply:
Problem is with created at and updted at value as it cannot be compared

Reply:
done